File: unity/Assets/Scripts/Multiplayer/Matchmaking/Requests/MatchmakingRequestInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking.Requests
{
    /// <summary>
    /// Detailed information about a matchmaking join request.
    /// Extends MatchmakingRequestBase with response details.
    /// </summary>
    [System.Serializable]
    public class MatchmakingRequestInfo : MatchmakingRequestBase
    {
        /// <summary>
        /// The ID of the player who responded to the request.
        /// </summary>
        public int responded_by;

        /// <summary>
        /// The name of the player who responded to the request.
        /// </summary>
        public string responder_name;

        /// <summary>
        /// Whether the matchmaking lobby requires approval for joining.
        /// </summary>
        public bool join_by_requests;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/requests/MatchmakingRequestInfo.java
package com.michitai.multiplayer.matchmaking.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.matchmaking.MatchmakingRequestBase;

/**
 * Information about a matchmaking join request, including responder details.
 */
public class MatchmakingRequestInfo extends MatchmakingRequestBase {
    @JsonProperty("responded_by")
    private Integer respondedBy;

    @JsonProperty("responder_name")
    private String responderName;

    @JsonProperty("join_by_requests")
    private boolean joinByRequests;

    public Integer getRespondedBy() {
        return respondedBy;
    }

    public void setRespondedBy(Integer respondedBy) {
        this.respondedBy = respondedBy;
    }

    public String getResponderName() {
        return responderName;
    }

    public void setResponderName(String responderName) {
        this.responderName = responderName;
    }

    public boolean isJoinByRequests() {
        return joinByRequests;
    }

    public void setJoinByRequests(boolean joinByRequests) {
        this.joinByRequests = joinByRequests;
    }
}
